source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/admin/settings/organizations/common/form.tsx

```diff
@@ -0,0 +1,279 @@
+import * as React from 'react';
+import { compose } from 'recompose';
+import { Checkbox, Dropdown } from 'semantic-ui-react';
+import { withTemplateHelpers, Mut, Toggle } from 'react-action-decorators';
+
+import { withTranslations, i18nProps } from '@lib/i18n';
+
+import {
+  query,
+  buildOptions,
+  withLoader,
+  attributesFor,
+  UserResource,
+  OrganizationResource
+} from '@data';
+
+
+interface IOwnProps {
+  organization: OrganizationResource;
+  owner: UserResource;
+  users: UserResource[];
+  onSubmit: (attributes: IState, onSuccess?: () => void) => void;
+  onCancel: () => void;
+}
+
+interface IState {
+  name?: string;
+  websiteUrl?: string;
+  owner?: UserResource;
+  ownerError?: string;
+  buildEngineUrl?: string;
+  buildEngineApiAccessToken?: string;
+  logoUrl?: string;
+  publicByDefault?: boolean;
+}
+
+type IProps =
+  & i18nProps
+  & IOwnProps;
+
+
+@withTemplateHelpers
+class OrganizationForm extends React.Component<IProps, IState> {
+  mut: Mut;
+  toggle: Toggle;
+
+  constructor(props: IProps) {
+    super(props);
+
+    const { organization, owner } = props;
+
+    if (organization) {
+
+      const {
+        name, websiteUrl, logoUrl,
+        buildEngineUrl, buildEngineApiAccessToken,
+        publicByDefault
+      } = attributesFor(organization);
+
+      this.state = {
+        name,
+        websiteUrl,
+        buildEngineUrl,
+        buildEngineApiAccessToken,
+        logoUrl,
+        publicByDefault,
+        owner
+      };
+
+    } else {
+      this.state = {
+        name: '',
+        websiteUrl: '',
+        buildEngineUrl: '',
+        buildEngineApiAccessToken: '',
+        logoUrl: '',
+        publicByDefault: false,
+        owner: null,
+        ownerError: ''
+      };
+    }
+
+  }
+
+  isValidForm = () => {
+    const { owner } = this.state;
+    if (!owner) {
+      this.setState({ ownerError: 'Cannot be empty' });
+    } else {
+      this.setState({ ownerError: '' });
+    }
+    return owner;
```

Comment:
why is the owner returned?

Reply:
Because I did wanted to do owner! == null or something like that but I can change it

Reply:
your method name asks a question, should the return value should be truthy

Reply:
changed